assembl/static2/js/app/components/administration/discussion/languageAdmin.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { I18n } from 'react-redux-i18n';
+import { compose, graphql } from 'react-apollo';
+import { connect } from 'react-redux';
+import { Checkbox } from 'react-bootstrap';
+
+import SectionTitle from '../sectionTitle';
+import withLoadingIndicator from '../../common/withLoadingIndicator';
+import getAllPreferenceLanguage from '../../../graphql/AllLanguagePreferences.graphql';
+
+
+class LanguageSection extends React.Component {
+
+  componentWillReceiveProps(nextProps){
+    const currentLocale = this.props.i18n.locale;
+    const nextLocale = nextProps.i18n.locale;
+    if (currentLocale !== nextLocale) {
+      //Refresh cache with new information, as the translated names are returned from backend
+      this.props.data.refetch({variables: {inLocale: nextLocale }});
```

Comment:
Great, I didn't know about refetch method, but according to http://dev.apollodata.com/react/receiving-updates.html our use case does not seem to be one of the recommended ones. So maybe there is a better solution? As we get data for another locale, I would say that we add data about this other locale to our store, instead of replacing the current one.